File: leetcode98_Validate Binary Search Tree_最大最小值_(AC).cpp
class Solution {
public:
    bool isValidBST(TreeNode* root) {
    	return min_max_judge(root,INT_MIN,INT_MAX);
    }
    bool min_max_judge(TreeNode* root,int min_value, int max_value){
    	if(root == NULL) return true;
    	if(root->val < max_value && root->val > min_value && (min_max_judge(root->left,min_value,root->val) && min_max_judge(root->right,root->val, max_value)))
    		return true;
    	else
    		return false;
    }

}
//不过

File: leetcode98_Validate Binary Search Tree_最大最小值_(AC).py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class Solution:  
    def isValidBST(self, root):  
    	return self.min_max_search(root,-2**31-1,2**31)#注意测试样例
    def min_max_search(self, root, min, max):
    	if root is None:
    		return True
    	return root.val < max and root.val > min and self.min_max_search(root.left,min,root.val) and self.min_max_search(root.right,root.val,max)